distributed/worker_memory.py

```diff
@@ -302,32 +313,46 @@ def __init__(
             dask.config.get("distributed.worker.memory.monitor-interval"),
             default=None,
         )
+        self.nanny = weakref.ref(nanny)
         assert isinstance(self.memory_monitor_interval, (int, float))
         if self.memory_limit and self.memory_terminate_fraction is not False:
             pc = PeriodicCallback(
-                partial(self.memory_monitor, nanny),
+                self.memory_monitor,
                 self.memory_monitor_interval * 1000,
             )
             nanny.periodic_callbacks["memory_monitor"] = pc
 
-    def memory_monitor(self, nanny: Nanny) -> None:
+    def get_process_memory(self) -> int:
+        """Get a measure for process memory.
+        This can be a mock target.
+        """
+        nanny = self.nanny()
+        if nanny:
+            try:
+                proc = nanny._psutil_process
+                return proc.memory_info().rss
+            except (ProcessLookupError, psutil.NoSuchProcess, psutil.AccessDenied):
+                return -1  # pragma: nocover
+        return -1
+
+    def memory_monitor(self) -> None:
         """Track worker's memory. Restart if it goes above terminate fraction."""
-        if nanny.status != Status.running:
-            return  # pragma: nocover
-        if nanny.process is None or nanny.process.process is None:
-            return  # pragma: nocover
-        process = nanny.process.process
-        try:
-            proc = nanny._psutil_process
-            memory = proc.memory_info().rss
-        except (ProcessLookupError, psutil.NoSuchProcess, psutil.AccessDenied):
+        nanny = self.nanny()
+        if (
+            not nanny
+            or nanny.status != Status.running
+            or nanny.process is None
+            or nanny.process.process is None
+            or self.memory_limit is None
+        ):
```

Comment:
```suggestion
        ):        
            assert self.memory_limit is not None
            assert self.memory_terminate_fraction is not False
```

Reply:
A few lines below we're calculating `memory / self.memory_limit`, i.e. if memory_limit is None we can skip `memory_monitor` / return early.


So, why remove the `self.memory_limit is None` from the if clause and why add these asserts? If these asserts are a concerning edge case, we should probably rather add a unit test

Reply:
it's not an edge case; the whole method will never be scheduled if memory_limit is disabled. See lines ~140 in ``__init__``. There are already several tests for it.